app/lib/provider_discovery.rb

```diff
@@ -22,6 +22,7 @@ def discover_provider(url, options = {})
         format ||= :xml if provider_endpoint
       end
 
+      raise OEmbed::NotFound, url unless provider_endpoint
```

Comment:
I think that it is better to be like `raise OEmbed::NotFound, 'URL is nil' if provider_endpoint.nil?`.

`raise` can only returns `String`